src/scrapers/base-scraper-with-browser.ts

```diff
@@ -291,7 +294,21 @@ class BaseScraperWithBrowser<TCredentials extends ScraperCredentials> extends Ba
       });
     }
 
-    await Promise.all(this.cleanups.reverse().map(cleanup => cleanup()));
+    for (const [i, cleanup] of this.cleanups.reverse().entries()) {
```

Comment:
@EranKoala hi :) as @baruchiro is going to merge #968 - you can revert the changes of the base scraper with browser and keep only the update of the scraper to avoid collision during merge